cdap-watchdog/src/main/java/io/cdap/cdap/metrics/process/MessagingMetricsProcessorService.java

```diff
@@ -401,15 +410,16 @@ private long processMetrics() {
         tryPersist();
 
         long endTime = System.currentTimeMillis();
-        // use currentMessageId != null to ensure that the current fetching is not empty and
-        // lastMetricTimeSecs is updated
-        if (currentMessageId != null && endTime - TimeUnit.SECONDS.toMillis(lastMetricTimeSecs) > maxDelayMillis) {
+        if (gotMessages && endTime - TimeUnit.SECONDS.toMillis(lastMetricTimeSecs) > maxDelayMillis) {
```

Comment:
I see that while initializing this variable we get a value in millis and convert and store it in secs. Here we converting from secs back to millis. If this variable isn't used elsewhere can we avoid the back-and-forth conversion?

Reply:
Now we have 4 usages of this value:
 * line 315, millis (added) - first initialization, mostly a safe thing in case we can't push any message after thread is started
 * line 391, seconds - main place it's updated
 * line 396, seconds - used to call updateTopicProcessingStats
 * line 413, millis - check if we a late
So it's 50%/50% seconds/millis. Generally this classes are a bit messy in using seconds vs millis but I don't want to make a refactoring to make it all millis